packages/react/src/analytics/integrations/Vitorino/Vitorino.ts

```diff
@@ -72,204 +67,87 @@ export default class Vitorino extends integrations.Integration<VitorinoIntegrati
    * @param analytics - Analytics stripped down instance.
    */
   constructor(
-    options: VitorinoIntegrationOptions,
+    options: VitorinoIntegrationOptions = {},
     loadData: LoadIntegrationEventData,
     analytics: StrippedDownAnalytics,
   ) {
+    const baseOptions: VitorinoIntegrationBaseOptions = omit(options, [
+      'riskified',
+      'forter',
+    ]);
     super(options, loadData, analytics);
 
-    this.isScriptLoaded = false;
+    this.baseOptions = baseOptions;
+    this.options = options;
 
-    this.getEventsMapper = GET_EVENTS_MAPPER_FN;
+    baseOptions.environment = getEnvironmentFromOptions(options);
 
-    this.isVitorinoConfigured = false;
-
-    this.vitorinoTrackCallback = null;
-
-    this.initializePromiseResolve = null;
-
-    this.userIdPromiseResolve = null;
-
-    this.config = null;
-
-    this.initializePromise = new Promise(resolve => {
-      this.initializePromiseResolve = resolve;
-    });
-
-    this.userIdPromise = new Promise(resolve => {
-      this.userIdPromiseResolve = resolve;
-    });
-
-    this.buildMapper(options);
-    this.loadScript(options);
-  }
-
-  /**
-   * When the user ID is available, config the vitorino and resolve the Promise so
-   * the tracking can begin.
-   *
-   * @param data - Analytics' event data.
-   */
-  override async onSetUser(data: SetUserEventData): Promise<void> {
-    await this.initializePromise;
-
-    if (data.user?.id) {
-      try {
-        this.configVitorino(data);
-
-        if (this.userIdPromiseResolve) {
-          this.userIdPromiseResolve();
-          this.userIdPromiseResolve = null;
-        }
-      } catch (error) {
-        utils.logger.error(
-          `${ERROR_MESSAGE_PREFIX} There was an error when trying to config Vitorino: ${error}`,
-        );
-      }
-    }
-  }
-
-  /**
-   * Makes the first call to Vitorino to configure it and receive the callback
-   * function to be called later when a page change occurs.
-   *
-   * @param data - Analytics' setUser event data.
-   */
-  configVitorino(data: SetUserEventData) {
-    this.config = this.buildConfigData(data);
-
-    this.vitorinoTrackCallback = window.Vitorino.track(this.config);
-
-    this.isVitorinoConfigured = true;
-  }
-
-  /**
-   * Prepares the mapper for the integration to work with.
-   *
-   * @param options - Integration options.
-   */
-  buildMapper(options: VitorinoIntegrationOptions) {
-    const customMapperFn = options.eventsMapper;
-
-    if (customMapperFn && typeof customMapperFn !== 'function') {
-      utils.logger.error(INVALID_CUSTOM_MAPPER_ERROR_MESSAGE);
+    // load default if not set
+    options.activeIntegrations ??= [
```

Comment:
Careful, you are setting a property in an object that is being shared with the code that called `addIntegration`. What do you think of setting it as a prop of the class?

Reply:
(or local variable)

Reply:
Was created a new variable for activeIntegrations instead of using previous from options directly.